### PR TITLE
hxtools: init at 2023122

### PR DESCRIPTION
**Pull request by meator**
## Description of changes

`hxtools` is an accumulated tool collection of J.Engelhardt: https://inai.de/projects/hxtools/

This is my first Nix package by the way. I was curious about Nix for some time now, so I wanted to try to package something. I'm open to all feedback.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [-] Tested basic functionality of all binary files (usually in `./result/bin/`) _(tested most binaries)_
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hx/hxtools/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    zstd
+  ];
+
+  buildInputs = [
+    libHX
+  ];
+
+  strictDeps = true;
+
+  meta = {
```
Comment (zimward) [inline pkgs/by-name/hx/hxtools/package.nix:30]: is there a reason you didn't add yourself to `maintainers`?

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -13167,6 +13167,13 @@
     githubId = 191622;
     name = "Denys Pavlov";
   };
+  meator = {
+    email = "meator.dev@gmail.com";
+    github = "meator";
+    githubId = 67633081;
+    name = "meator";
+    keys = [ { fingerprint = "7B0F 58A5 E0F1 A2EA 1157  8A73 1A14 CB34 64CB E5BF"; } ];
```
Comment (zimward) [inline maintainers/maintainer-list.nix:13175]: please sign your commit with that key.

**Review by zimward — CHANGES_REQUESTED** [review]
apart from my remarks everything looks fine and works as expected.

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (meator) [reply maintainers/maintainer-list.nix:13175]: Ah, I forgot to enable gpgsign in the clone.

File: pkgs/by-name/hx/hxtools/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    homepage = "https://inai.de/projects/hxtools/";
+    description = "Collection of small tools over the years by j.eng";
+    # Taken from https://codeberg.org/jengelh/hxtools/src/branch/master/LICENSES.txt
+    license = with lib.licenses; [
+      mit
+      bsd2Patent
+      lgpl21Plus
+      gpl2Plus
+    ];
+    maintainers = with maintainers; [ meator ];
```
Comment (zimward) [inline pkgs/by-name/hx/hxtools/package.nix:40]: did you mean `lib.maintainers`?

**Review by zimward — CHANGES_REQUESTED** [review]

**Review by zimward — APPROVED** [review]

**Comment by meator** [comment]
@zimward Thanks for the review!

Also, is the failed check related to my package?

**Comment by zimward** [comment]
i am not entirely sure what to make of the ofborg output. `nixpkgs-review` took quite some time (and memory) to evaluate on my machine, maybe that is why.

**Comment by qubitnano** [comment]
@ofborg eval

File: pkgs/by-name/hx/hxtools/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  libHX,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hxtools";
+  version = "20231224";
+
+  src = fetchurl {
+    url = "https://inai.de/files/hxtools/hxtools-${finalAttrs.version}.tar.zst";
+    sha256 = "sha256-TyT9bsp9qqGKQsSyWCfd2lH8ULjqJ5puMTw2TgWHV5c=";
```
Comment (qubitnano) [inline pkgs/by-name/hx/hxtools/package.nix:16]:
```suggestion
    hash = "sha256-TyT9bsp9qqGKQsSyWCfd2lH8ULjqJ5puMTw2TgWHV5c=";
```

File: pkgs/by-name/hx/hxtools/package.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    # Taken from https://codeberg.org/jengelh/hxtools/src/branch/master/LICENSES.txt
+    license = with licenses; [
+      mit
+      bsd2Patent
+      lgpl21Plus
+      gpl2Plus
+    ];
+    maintainers = with maintainers; [ meator ];
+    platforms = platforms.all;
+  };
```
Comment (qubitnano) [inline pkgs/by-name/hx/hxtools/package.nix:42]:
```suggestion
  meta = {
    homepage = "https://inai.de/projects/hxtools/";
    description = "Collection of small tools over the years by j.eng";
    # Taken from https://codeberg.org/jengelh/hxtools/src/branch/master/LICENSES.txt
    license = with lib.licenses; [
      mit
      bsd2Patent
      lgpl21Plus
      gpl2Plus
    ];
    maintainers = with lib.maintainers; [ meator ];
    platforms = lib.platforms.all;
  };
```

Comment (meator) [reply pkgs/by-name/hx/hxtools/package.nix:42]: Why?

Comment (zimward) [reply pkgs/by-name/hx/hxtools/package.nix:42]: use of `with lib;` is considered as an anti-pattern. See #208242

Comment (zimward) [reply pkgs/by-name/hx/hxtools/package.nix:42]: no worries. i also hadn't had it on my mind when reviewing and only stumbled over it in another pr.

**Pull request merged**